pkgs/by-name/gk/gk-cli/package.nix

```
@@ -28,18 +29,49 @@ stdenv.mkDerivation (finalAttrs: {
   installPhase = ''
     runHook preInstall
 
-    install -Dm555 gk -t $out/bin/
+    install -Dm555 gk*/gk -t $out/bin/
 
-    installShellCompletion --bash ./**/gk.bash
-    installShellCompletion --fish ./**/gk.fish
-    installShellCompletion --zsh ./**/_gk
+    wrapProgram $out/bin/gk \
+      --prefix PATH : "${lib.makeBinPath [ gitMinimal ]}"
 
     runHook postInstall
   '';
 
-  postFixup = ''
-    wrapProgram $out/bin/gk \
-      --prefix PATH : "${lib.makeBinPath [ coreutils ]}"
+  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    generate_completion() {
+      local TARGET_SHELL="$1"
+      local TEMPORARY_HOME="$(mktemp --directory)"
+
+      # Create an empty config file to avoid "no config file found" at the beggining of generated completion
+      local CONFIG_FILE="$TEMPORARY_HOME/.gk/cli/gk_config.yaml"
+      mkdir --parents "$(dirname "$CONFIG_FILE")"
+      touch "$CONFIG_FILE"
+
+      installShellCompletion --cmd gk \
+        "--$TARGET_SHELL" <(
+          HOME="$TEMPORARY_HOME" \
+          timeout 3 `# Use timeout because gk hangs instead of closing in the sandbox` \
+          $out/bin/gk completion "$TARGET_SHELL"
+        )
+    }
+
+    generate_completion bash
+    generate_completion fish
+    generate_completion zsh
```

the `generate_completion` becomes a bit complex to read when it controls everything, but a simple `mkhome` is quicker to grasp. How about
```suggestion
    # We use timeout because gk hangs instead of closing in the sandbox
    # We use 3 different homes since the timeouts make gk/dgraph/badger create corrupt caches
    mkhome() {
      TMP_HOME=$(mktemp --directory)
      # Create empty config file to avoid "no config file found" errors
      mkdir --parents "$TMP_HOME/.gk/cli/"
      touch "$TMP_HOME/.gk/cli/gk_config.yaml"
      echo "$TMP_HOME"
    }
    installShellCompletion --cmd gk \
      --bash <(HOME="$(mkhome)" timeout 3 $out/bin/gk completion bash) \
      --fish <(HOME="$(mkhome)" timeout 3 $out/bin/gk completion fish) \
      --zsh <(HOME="$(mkhome)" timeout 3 $out/bin/gk completion zsh)

```

---

Since the last version, creating config is not required anymore

I have re-inlined home creation